rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,225 @@
+
+It is also suggested in this RFC that people, upon marking licenses as `runnableOnHydra`, check all the derivations that use this license.
+They could then have to mark them as either `hydraPlatforms = []`, `preferLocalBuild = true` and/or `allowSubstitutes = false`.
+This might be useful for packages like TPTP:
+they may not yet be marked as such due to these flags having no impact on unfree packages;
+but would take gigabytes on Hydra for basically no local build time improvement
+
+With this in mind, Hydra could start building, among others:
+- CUDA
+- DragonflyDB
+- MongoDB
+- Nomad
+- NVIDIA drivers
+- Outline
+- SurrealDB
+- TeamSpeak
+- Terraform
+- Unrar
+- Vagrant
+- NixOS tests that involve such software (eg. MongoDB or Nomad)
+
+And Hydra will keep not building, among others:
+- CompCert
+- DataBricks
+- Elasticsearch
+- GeoGebra
+- Widevine CDM
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Technical drawbacks
+
+The main risk is that NixOS could end up including unfree software in an installation image if:
+1. we forgot to add it to the list of no-allowed-unfree jobs, and
+2. a maintainer did actually add unfree software to that build.
+
+This seems exceedingly unlikely, making this change basically risk-free.
+
+The only remaining drawback is that Hydra would have to evaluate Nixpkgs twice, thus adding to eval times.
+However, the second eval (with no-unfree) should be reasonably small and not actually evaluate all packages, as it is only used for installation media.
+
+## Political drawbacks
+
+Whether distributing unfree software is a political drawback is left to each reader's opinion.
+
+Besides that, there are three main political risks.
+First is, this RFC could end up completely unused.
+Maybe, with proper license investigation, we will notice that none of the packages listed above can actually be redistributed by Hydra.
+
+The second risk is one of manpower.
+We may need the Foundation's input on whether a specific license is ok to redistribute or not.
+This could require some manpower from the Foundation's side.
+
+Finally, the third risk is one of propagation.
+With both hydra and some nixos maintainers running with `allowUnfree`, there is a risk that free packages start unnecessarily depending on unfree packages.
+This would then break the setup of the people not actually running with `allowUnfree`.
+
+This being said, all these risks are probably less impactful than the current statu quo.
+Indeed, we currently have packages for Mac that are not marked with any license, because they would otherwise have to be marked unfree,
+yet we do want to build and test them.
+This means that we are already lying on licenses in order to get them through Hydra.
+And, in particular, this means they could actually reach the machine of users without `allowUnfree`.
+This situation is entirely due to the absence of this RFC, and could only be improved by it.
```

Comment:
I don’t think this is accurate, or at least it’s completely orthogonal to this RFC if so. I believe that the interface definitions we use from the SDK are either uncopyrightable or effectively so due to falling under *Google v. Oracle* interoperability, depending on the jurisdiction, but if we were to reject that argument then they would not be marked as redistributable at all. In that case, continuing to provide macOS support would require investing in Hydra/cache engineering work so that every builder could download the SDK separately without being cached, and this RFC would not signal any approval to building it. Therefore this RFC changes absolutely nothing about the situation. I think this should probably just be dropped entirely.

Reply:
Even if assuming Google v. Oracle applies, what license would you set on the interface files? That's basically my line of thinking.

This being said, you do make a valid point, so I'll need to think more about this and come back to it.